### DISCUSSION TITLE
Type mismatch between Notification CC value and states map

### DISCUSSION THREAD

**Discussion by raman325** [zwave-js/zwave-js#1559] [Q&A] [answered]
While working on the HA integration, we noticed that the value for a Notification CC Value is an int, but in the metadata.states property, the keys are strings. Why is that?

**Comment by AlCalzone** [comment, marked as answer]
Oh, that's just JavaScript for you. In JS objects, numeric keys and their string counterparts are equivalent:
```
> var obj = {}
undefined
> obj[0] = "abc"
'abc'
> obj
{ '0': 'abc' }
> obj['0']
'abc'
> obj[0]
'abc'
```

At least for `node-zwave-js`, you can expect that a key that looks like a number is meant to be a number.

**Reply by raman325** [reply]
thanks for the clarification!
